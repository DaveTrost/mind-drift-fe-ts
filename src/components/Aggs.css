@import './vars.css';

.Aggs {
  display: flex;
  flex-direction: column;
  margin: 2em;
  h3 {
    text-align: center;
    font-family: Damion;
    font-weight: 700;
    font-size: 2em;
    margin-left: .4em;
    margin-bottom: 2em;
    color: var(--rust);
  }
  .individualBadges {
    display: flex; 
    font-family: Montserrat;
    margin-left: 2em;
    margin: 0;
    p {
      font-family: Montserrat;
      font-size: 1em;
      font-weight: 700;
      color: var(--dark-blue);
      margin-left: 2em;
    }
  }
  h1{
    color: var(--rust);
    font-weight: 1000;
    font-size: .7em;

  }
  h2{
    color: var(--rust);
    font-weight: 1000;
    font-size: .8em;
  }
  img {
    position: relative;
    z-index: -1;
    top: -3.9em;
  }
  
  .badge {
    display: flex;
    flex-direction: column;
    text-align: center;
    height: 5em;
  }
}


