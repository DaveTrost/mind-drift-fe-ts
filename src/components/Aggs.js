import React from 'react';
import PropTypes from 'prop-types';
import styles from './Aggs.css';
import blueicon from '../assets/blueicon.png';
import greenicon from '../assets/greenicon.png';
import peachicon from '../assets/peachicon.png';

const Aggs = ({ average, total, streak }) => {
  return (
    <div className={styles.Aggs}>
      <h3>here are your stats</h3>
      
      <p>
        <div className={styles.badge}>
          <h2>{average.time}</h2>
          <h1>{average.type}</h1>
          <img src={blueicon} />
        </div>
        average session
      </p>
     
      <p>
        <div className={styles.badge}>
          <h2>{total.time}</h2>
          <h1>{total.type}</h1>
          <img src={greenicon} />
        </div>
        total time breathing
      </p>
      
      <p>
        <div className={styles.badge}>
          <h2>{streak}</h2>
          <h1>days</h1>
          <img src={blueicon} />
        </div>
        current streak
      </p>
    </div>
  );
};

Aggs.propTypes = {
  average: PropTypes.object.isRequired,
  total: PropTypes.object.isRequired,
  streak: PropTypes.number.isRequired
};

export default Aggs;
