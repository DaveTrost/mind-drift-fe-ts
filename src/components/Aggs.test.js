import React from 'react';
import { shallow } from 'enzyme';
import Aggs from './Aggs';

describe('Aggregations component', () => {

  it('renders User Stats from Aggregation data', () => {
    const average = {
      time: 1.4,
      type: 'minutes',
    };
    const total = {
      time: 2.5,
      type: 'hours',
    };
    const wrapper = shallow(<Aggs average={average} total={total} streak={5} />);
    expect(wrapper).toMatchSnapshot();
  });
});
