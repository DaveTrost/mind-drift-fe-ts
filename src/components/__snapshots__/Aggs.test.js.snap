// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aggregations component renders User Stats from Aggregation data 1`] = `
<div
  className="Aggs"
>
  <h3>
    Your Stats:
  </h3>
  <section
    className="individualBadges"
  >
    <div
      className="badge"
    >
      <h2>
        1.4
      </h2>
      <h1>
        minutes
      </h1>
      <img
        src="test-file-stub"
      />
    </div>
    <p>
      Average Session
    </p>
  </section>
  <section
    className="individualBadges"
  >
    <div
      className="badge"
    >
      <h2>
        2.5
      </h2>
      <h1>
        hours
      </h1>
      <img
        src="test-file-stub"
      />
    </div>
    <p>
      Total Time Breathing
    </p>
  </section>
  <section
    className="individualBadges"
  >
    <div
      className="badge"
    >
      <h2>
        5
      </h2>
      <h1>
        days
      </h1>
      <img
        src="test-file-stub"
      />
    </div>
    <p>
      Current Streak
    </p>
  </section>
</div>
`;
